plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslPlugin.kt

```diff
@@ -38,7 +38,7 @@ open class KotlinDslPlugin : Plugin<Project> {
 
             applyEmbeddedKotlinPlugin()
             applyKotlinDslCompilerPlugins()
-            addGradleKotlinDslDependencyTo("compileOnly", "testRuntimeOnly")
+            addGradleKotlinDslDependencyTo("compileOnly", "testImplementation")
```

Comment:
What if a user isn't using the `java-library` plugin? Or is it automatically applied when this plugin is applied?

Reply:
I think the `embeddedKotlin` already applies the implementation somehow, right? 🤔

Reply:
@JLLeitschuh the `testImplementation` configuration is added by the [`java` plugin](https://docs.gradle.org/current/userguide/java_plugin.html), no need for the `java-library` plugin for it to exist. I think you mixed with the `api` configuration that is only available when using the `java-library` plugin.